eng/common/pipelines/templates/validate-filename.yml

```diff
@@ -0,0 +1,13 @@
+steps:
+- pwsh: |
+    $differByCaseFiles = git ls-files | Group-Object | Where-Object { $_.Count -gt 1 } | ForEach-Object { $_.Group} 
+
+    if ($differByCaseFiles) 
+    {
+      Write-Error "Do NOT name the files which only differ in case. Please check and rename the following files:"
+      $differByCaseFiles | Write-Error
+      exit 1
+    }
+    Write-Host "There are no files only differ in case. "
+  displayName: Validate on files only differ on cases
+  workingDirectory: $(System.DefaultWorkingDirectory)
```

Comment:
This working directory may end up needing to be a template parameter as we will need to make sure that this is ran at the root of cloned repo which can vary.